### ISSUE TITLE
Drop Python 2.4 support

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4962]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#4962|
|Type|enhancement|
|Created|2011-03-17 18:21:56Z|

RHEL 6 has been released, with a modern version of Python (RHEL 5 was the last holdout for 2.4). The compatible CentOS 6 isn't out yet, but presumably will be out soon.

I propose that Python 2.4 support be dropped in the next Twisted release after September -- Python 2.5 was released in Sep '06, so that's a 5 year window for people to upgrade. People who must stick with old distributions can build custom Python 2.5 package, they're already building Twisted as a custom package if they're using latest release.

If this is agreed, Twisted 11 should note this in the NEWS file. Jean-Paul additionally suggests a timeline on the wiki.

James proposes extending this to 2.6, since modern distributions have that.

<details><summary>Searchable metadata</summary>

```
trac-id__4962 4962
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Twisted_11_1 Twisted-11.1
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1300386116000000 1300386116000000
changetime__1323378145000000 1323378145000000
version__None None
owner__therve therve
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

Er, James proposes we drop 2.5 as well.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

I'm adding this to the 11.2 milestone, which will be the first release after September.  We didn't announce this in 11.0, but we can still do it in 11.1, which will be in July.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

The Python 2.3 reference in [twisted/topfiles/setup.py](twisted/topfiles/setup.py) also needs to be fixed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#note_5):
> The Python 2.3 reference in [twisted/topfiles/setup.py](twisted/topfiles/setup.py) also needs to be fixed.
Opened [#5060](https://github.com/twisted/twisted/issues/5060) for this.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml commented|
|-|-|

I think it's a good idea to release an 11.1 that supports the same Pythons as 11.0 and announces that it's the last such release.

Note that this is not a felt need for Canonical. We only care about the version of Python that's in the latest Ubuntu LTS, Python 2.6.

(Also, I don't think it matters much how big the delta is between 11.1 and 11.2, as long as it's non-zero.)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst set owner to @therve|
|-|-|

Changing to milestone 11.1 since this has to be announced with 11.1.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

11.1 was released and the [milestone:Twisted-11.1 milestone] should be closed.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set status to `closed`|
|-|-|
